src/main/java/com/opendxl/databus/consumer/DatabusPushConsumer.java

```diff
@@ -0,0 +1,442 @@
+            // Prevents to logging  if a WakeupException was thrown because of calling close
+            if ((e.getCause() instanceof WakeupException)  && stopRequested.get()) {
+                LOG.error("Consumer " + super.getClientId() + "Error: " + e.getMessage(), e);
+            }
+        } catch (Exception e) {
+            LOG.error("Consumer " + super.getClientId() + "Error: " + e.getMessage(), e);
+        } finally {
+            countDownLatch.countDown();
+            LOG.info("Consumer " + super.getClientId() + " end");
+        }
+
+    }
+
+    /**
+     * It sets Consumer position
+     *
+     * @param consumerPosition A Map with the offset position per topic-partition
+     */
+    private void seek(final Map<TopicPartition, Long> consumerPosition) {
+        for (TopicPartition tp : super.assignment()) {
+            if (consumerPosition.containsKey(tp)) {
+                final Long position = consumerPosition.get(tp);
+                super.seek(tp, position);
+            }
+        }
+    }
+
+    /**
+     * It spawns a thread to execute the listener
+     *
+     * @param databusPushConsumerFuture The future that track the Pusg COnsumer
+     * @param records records read from Databus
+     * @return a Future instance to
+     */
+    private Future<DatabusPushConsumerListenerResponse> runListenerAsync(
+            final DatabusPushConsumerFuture databusPushConsumerFuture,
+            final ConsumerRecords records) {
+
+        // launch the listener in a separate thread
+        Future<DatabusPushConsumerListenerResponse> listenerTaskFuture = executor.submit(
+                () -> consumerListener.onConsume(records));
+
+        // Update the Push Consumer status
+        databusPushConsumerFuture
+                .setDatabusPushConsumerListenerStatus(
+                        new DatabusPushConsumerStatus.Builder()
+                                .withStatus(DatabusPushConsumerStatus.Status.PROCESSING)
+                                .build());
+
+        LOG.info("Consumer " + getClientId() + " Listener was called");
+        return listenerTaskFuture;
+    }
+
+
+    /**
+     * It stores the current consumer position based on the partitions assigned.
+     * This is the offset position per topicPartition before polling messages from Databus.  Position
+     * will be used to rewind the consumer in case the listener returns RETRY enum value.
+     *
+     * @return The map with the current offset position for each topic partition assigned to consumer
+     */
+    private Map<TopicPartition, Long> getCurrentConsumerPosition(final Set<TopicPartition> assignment) {
+        final Map<TopicPartition, Long> lastKnownPositionPerTopicPartition = new HashMap(assignment.size());
+        for (TopicPartition tp : assignment) {
+            lastKnownPositionPerTopicPartition.put(tp, super.position(tp));
+        }
+        return lastKnownPositionPerTopicPartition;
+    }
+
+    @Override
```

Comment:
Should the inherit javadoc be present here?

    /**
     * {@inheritDoc}
     */